samples/browser/src/main/java/org/mozilla/samples/browser/addons/AddOnsFragment.kt

```diff
@@ -117,9 +159,12 @@ class AddOnsFragment : Fragment(), View.OnClickListener {
 
             holder.titleView.text = addOn.translatableName.translate()
             holder.summaryView.text = addOn.translatableSummary.translate()
-            holder.view.tag = addOn
-            holder.view.setOnClickListener(clickListener)
-            holder.addButton.setOnClickListener(clickListener)
+            holder.itemView.tag = addOn
+            holder.itemView.setOnClickListener(clickListener)
+            holder.addButton.isVisible = !addOn.installed
```

Comment:
Good catch :)
It will be ideal to do it in that way, but but unfortunately,  as [RecyclerView re-uses views](https://developer.android.com/reference/androidx/recyclerview/widget/RecyclerView.Adapter.html?hl=en#onBindViewHolder(VH,%20int)) I have to make sure that this code always gets executed and If I put the conditions together it could lead to some cases where the visibility of the view doesn't get updated if the condition is not true.

You made me realized that I also have to set the `setOnClickListener` to `null` to prevent it keeping previous references when it's not need it.

```kotlin
            val listener = if (!addOn.installed) {
                clickListener
            } else {
                null
            }
            holder.addButton.setOnClickListener(listener)
```